pkg/util/net/net.go

```diff
@@ -72,3 +89,165 @@ func setBuffers(rc syscall.RawConn, sz int) error {
 
 	return err
 }
+
+type PrivateZoneDeletionConfig struct {
+	Log                *logrus.Entry
+	PrivateZonesClient privatedns.PrivateZonesClient
+	Configcli          configclient.Interface
+	Mcocli             mcoclient.Interface
+	Kubernetescli      kubernetes.Interface
+	VNetLinksClient    privatedns.VirtualNetworkLinksClient
+	ResourceGroupID    string
+}
+
+func RemovePrivateDNSZone(ctx context.Context, config PrivateZoneDeletionConfig) error {
+	if config.PrivateZonesClient == nil {
+		return errors.New("privateZonesClient is nil")
+	}
+
+	resourceGroup := stringutils.LastTokenByte(config.ResourceGroupID, '/')
+	privateZones, err := config.PrivateZonesClient.ListByResourceGroup(ctx, resourceGroup, nil)
+	if err != nil {
+		config.Log.Print(err)
+		return nil
+	}
+
+	if config.Configcli == nil {
+		return errors.New("configcli is nil")
+	}
+
+	if len(privateZones) == 0 {
+		// fix up any clusters that we already upgraded
+		if err := UpdateDNSes(ctx, config.Configcli.ConfigV1().DNSes(), config.ResourceGroupID); err != nil {
+			config.Log.Print(err)
+		}
+		return nil
+	}
+
+	if config.Mcocli == nil {
+		return errors.New("mcocli is nil")
+	}
+
+	if config.Kubernetescli == nil {
+		return errors.New("kubernetescli is nil")
+	}
+
+	mcpLister := config.Mcocli.MachineconfigurationV1().MachineConfigPools()
+	nodeLister := config.Kubernetescli.CoreV1().Nodes()
+	sameNumberOfNodesAndMachines, err := ready.SameNumberOfNodesAndMachines(ctx, mcpLister, nodeLister)
+	if err != nil {
+		config.Log.Print(err)
+		return nil
```

Comment:
Should we be returning the err here? Same applies to the rest of the return nils under this block

Reply:
Good question. The thing is that we weren't doing that in the original code and here I just want to keep the same behavior. Even if returning nil when there is an error is counterintuitive (it is for sure!), I should keep the logic the same way it was before the PR.

Reply:
Sure thing, I didn't really check to see what changes you made from the original, just something I noticed :) Should we cut a card to ensure this doesn't get lost?